tests/unit/select_test.coffee

```diff
@@ -344,3 +344,22 @@ test 'optionValuePath with nested valuePath', ->
 
   equal obj2, select.get('selection'),
     'The right selection is retrieved'
+
+test 'shouldEnsureVisible controls whether to ensure visibility', ->
+  expect 2
+
+  select = @subject
+    content: ['foo']
+    ensureVisible: ->
+      ok yes, 'ensureVisible is called if shouldEnsureVisible is true'
+  select.set 'highlighted', 'foo'
+
+  select = @subject
+    content: ['bar']
+    shouldEnsureVisible: no
+
+  spy = sinon.spy select, 'ensureVisible'
+  select.set 'highlighted', 'bar'
+  equal spy.callCount, 0,
+    'ensureVisible is not called if shouldEnsureVisible is false'
+  spy.restore()
```

Comment:
I pasted the link is just to show the global variable usage. What I said above is a good practice is to restore it in teardown. It's not necessary that each test needs at most one spy, but you can always reuse it, if not you can have multiple of them. However, it is fine in this case because the spy is on the `select` component which will be destroy between tests.